src/OrbitGl/ThreadTrack.cpp

```diff
@@ -69,22 +68,27 @@ std::string ThreadTrack::GetThreadNameFromTid(uint32_t thread_id) {
   ;
 }
 
-void ThreadTrack::InitializeNameAndLabel() {
+const std::string_view kAllThreads = " (all_threads)";
+const std::string ThreadTrack::GetLabel() const {
+  auto thread_id = GetThreadId();
+  auto name = GetName();
+  if (thread_id == orbit_base::kAllThreadsOfAllProcessesTid) {
+    return name;
+  } else if (thread_id == orbit_base::kAllProcessThreadsTid) {
+    return name.append(kAllThreads);
+  }
+  std::string tid_str = std::to_string(thread_id);
+  return absl::StrFormat("%s [%s]", name, tid_str);
+}
+
+int ThreadTrack::GetNumberOfPrioritizedTrailingCharacters() const {
+  auto thread_id = GetThreadId();
   if (GetThreadId() == orbit_base::kAllThreadsOfAllProcessesTid) {
-    SetLabel("All tracepoint events");
+    return 0;
   } else if (GetThreadId() == orbit_base::kAllProcessThreadsTid) {
-    // This is the process track.
-    std::string process_name = capture_data_->process_name();
-    const std::string_view all_threads = " (all_threads)";
-    SetLabel(process_name.append(all_threads));
-    SetNumberOfPrioritizedTrailingCharacters(all_threads.size() - 1);
-  } else {
-    const std::string& thread_name = GetThreadNameFromTid(GetThreadId());
-    std::string tid_str = std::to_string(GetThreadId());
-    std::string track_label = absl::StrFormat("%s [%s]", thread_name, tid_str);
-    SetLabel(track_label);
-    SetNumberOfPrioritizedTrailingCharacters(tid_str.size() + 2);
+    return kAllThreads.size() - 1;
   }
+  return std::to_string(thread_id).size() + 2;
```

Comment:
Could you add comments to explain these two cases? Maybe with an example how to shortened string will look like? Otherwise to code is quite hard to understand

Reply:
This is the problem when having separated Label with NumberOfPrioritizedTrailingCharacters, maybe a comment could help, thanks!

Reply:
I've added comments, let me know what do you think about that

Reply:
I added two extra comments to the comments. :sweat_smile: So I'm closing this thread.